parts/k8s/cloud-init/artifacts/cse_install.sh

```diff
@@ -245,13 +246,26 @@ extractHyperkube() {
     fi
 }
 
+extractKubeBinaries() {
+    K8S_TGZ_TMP=$(echo ${KUBE_BINARY_URL} | cut -d "/" -f 5)
```

Comment:
this is called both by CSE at runtime and by the VHD to pre-install versions. We need a way to pre-install several versions of 1.17 (eg. 1.17.1 and 1.17.2) in the VHD and, at CSE runtime: 1) download the desired version if not present 2) if it's already present, use it without going out to the internet 3) remove all unused version binaries

Reply:
But when it's called in the `installKubeletAndKubectl` func below, it's guarded by a `[[ ! -f ]]` test that checks for the presence of the `kubectl-1.17.0` binary (for example). Only if the binary isn't already present do we go out to the internet to fetch it. The logic is parallel to how we use `extractHyperkube` so I think it's doing what we want.

Could be entirely mistaken, so apologies if I'm still off-base. Long day.

Reply:
My bad, you're completely right 🙏  I read the code too fast and forgot that `[[ ! -f ]]` was in `installKubeletAndKubectl` and not `extractHyperkube`...
So just to confirm, this function will create `/usr/local/bin/kubelet-${KUBERNETES_VERSION}` and `/usr/local/bin/kubectl-${KUBERNETES_VERSION}`?

Reply:
Yes, it does, so the binaries in the same place with the same names that `extractHyperkube` would have put them and we can reuse that bit of logic.

(The `tar --transform=` argument is what renames the binaries while extracting them. It's a neat trick but it's hard to grok the resulting command line.)

Reply:
Got it, in that case it's exactly what we want. 

VHD integration for kube binaries lgtm, let me know if you'd like another review of the whole PR.